### PR TITLE
[13.0][ADD] product_form_account_move_line_link

### PR DESCRIPTION
**Pull request by JaumeBforgeFlow**
Adds a smart button on product forms to access account move lines where the product is used.

Similar to https://github.com/OCA/purchase-workflow/pull/715 and https://github.com/OCA/sale-workflow/pull/897

### CODE REVIEW

File: product_form_account_move_link/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    account_move_lines_count = fields.Integer(
+        compute="_compute_account_move_lines_count", string="Move Lines"
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/models/product_product.py]:
```suggestion
        compute="_compute_account_move_lines_count", string="Invoiced"
```

File: product_form_account_move_link/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    account_move_lines_count = fields.Float(
+        compute="_compute_account_move_lines_count", string="Purchased"
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/models/product_template.py]:
```suggestion
        compute="_compute_account_move_lines_count", string="Invoiced"
```

File: product_form_account_move_link/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    account_move_lines_count = fields.Integer(
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/models/product_product.py]:
```suggestion
    account_move_lines_count = fields.Float(
```

File: product_form_account_move_link/views/product_product_views.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_product_views.xml]: Correct copyright

File: product_form_account_move_link/views/product_template_views.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_template_views.xml]: Correct copyright

File: product_form_account_move_link/views/product_template_views.xml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        <field name="arch" type="xml">
+            <div name="button_box" position="inside">
+                <button
+                    class="oe_stat_button"
+                    name="%(product_form_account_move_link.action_account_move_line_product_tree)d"
+                    type="action"
+                    icon="fa-list"
+                >
+                    <field
+                        string="Account Moves"
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_template_views.xml]:
```suggestion
                        string="Journal Items"
```

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.actions.act_window" id="action_account_move_line_product_tree">
+        <field name="context">{'search_default_approved':1}</field>
+        <field name="domain">[('product_id.product_tmpl_id','in',active_ids)]</field>
+        <field name="name">Account Move Lines</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]:
```suggestion
        <field name="name">Journal Items</field>
```

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        <field name="res_model">account.move.line</field>
+        <field name="view_id" ref="account.view_move_line_tree" />
+    </record>
+    <record
+        model="ir.actions.act_window"
+        id="action_account_move_line_product_product_tree"
+    >
+        <field name="context">{'search_default_approved':1}</field>
+        <field name="domain">[('product_id','in',active_ids)]</field>
+        <field name="name">Account Move Lines</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]:
```suggestion
        <field name="name">Journal Items</field>
```

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]: Correct copyright

File: product_form_account_move_link/views/product_product_views.xml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+<odoo>
+    <record id="product_normal_form_view_inherit" model="ir.ui.view">
+        <field name="name">product.product.account.move.button</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_product_views.xml]:
```suggestion
        <field name="name">product.product.form (in product_form_account_move_link)</field>
```

File: product_form_account_move_link/views/product_template_views.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="product_template_only_form_view_inherit" model="ir.ui.view">
+        <field name="name">product.template.account.move.button</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_template_views.xml]:
```suggestion
        <field name="name">product.template.form (in product_form_account_move_link)</field>
```

File: product_form_account_move_link/models/product_product.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        compute="_compute_account_move_lines_count", string="Move Lines"
+    )
+
+    def _compute_account_move_lines_count(self):
+        if not self.user_has_groups("account.group_account_user") or not self.ids:
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/models/product_product.py]:
```suggestion
        if not self.user_has_groups("account.group_account_invoice") or not self.ids:
```

I would suggest to use the group_account_invoice which is the base one for accounting

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 13]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 ForgeFlow S.L.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]:
```suggestion
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0) -->
```

File: product_form_account_move_link/views/product_template_views.xml [hunk 14]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 ForgeFlow S.L.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_template_views.xml]:
```suggestion
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0) -->
```

File: product_form_account_move_link/views/product_product_views.xml [hunk 15]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 ForgeFlow S.L.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/product_product_views.xml]:
```suggestion
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0) -->
```

File: product_form_account_move_link/models/product_template.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/models/product_template.py]: Why is _logger used?

**Comment by JordiMForgeFlow** [comment]
@JaumeBforgeFlow Readme folder is also missing

File: product_form_account_move_link/__manifest__.py [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+    "summary": """
+        Adds a button on product forms to access Journal Items""",
+    "version": "13.0.1.0.0",
+    "author": "ForgeFlow, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/account-invoicing",
+    "category": "Account",
+    "license": "AGPL-3",
+    "depends": ["account", "product"],
+    "data": [
+        "views/purchase_order_line_views.xml",
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/__manifest__.py]:
```suggestion
        "views/account_move_line_views.xml",
```

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 18]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 ForgeFlow S.L.
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0) -->
+<odoo>
+    <record model="ir.actions.act_window" id="action_account_move_line_product_tree">
+        <field name="context">{'search_default_approved':1}</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]:
```suggestion
        <field name="context">{'search_default_posted':1}</field>
```

File: product_form_account_move_link/views/purchase_order_line_views.xml [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+        <field name="domain">[('product_id.product_tmpl_id','in',active_ids)]</field>
+        <field name="name">Journal Items</field>
+        <field name="res_model">account.move.line</field>
+        <field name="view_id" ref="account.view_move_line_tree" />
+    </record>
+    <record
+        model="ir.actions.act_window"
+        id="action_account_move_line_product_product_tree"
+    >
+        <field name="context">{'search_default_approved':1}</field>
```
Comment (JordiMForgeFlow) [inline product_form_account_move_link/views/purchase_order_line_views.xml]:
```suggestion
        <field name="context">{'search_default_posted':1}</field>
```

**Review by JordiMForgeFlow — APPROVED** [review]
Code + functional review 👍🏼

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM 👌🏽

**Review by JoanSForgeFlow — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
Shouldn't it be product_form_account_move_line_link ?

**Comment by JordiMForgeFlow** [comment]
Hi @AaronHForgeFlow 

It is following the same convention as in https://github.com/OCA/sale-workflow/tree/13.0/product_form_sale_link and https://github.com/OCA/purchase-workflow/tree/13.0/product_form_purchase_link

**Comment by AaronHForgeFlow** [comment]
@JordiMForgeFlow ok! I prefer the other way but it is not big deal.

**Comment by JaumeBforgeFlow** [comment]
Hi @AaronHForgeFlow, i've changed the module name!

**Review by JordiMForgeFlow — APPROVED** [review]
Thanks for the improvement @JaumeBforgeFlow! 🚀

**Review by AaronHForgeFlow — APPROVED** [review]
Functional ok

**Pull request merged**